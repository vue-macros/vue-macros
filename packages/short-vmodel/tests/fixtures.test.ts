import { resolve } from 'node:path'
import {
  rollupBuild,
  RollupVue,
  testFixtures,
  UnpluginOxc,
} from '@vue-macros/test-utils'
import { describe } from 'vitest'
import ShortVmodel from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    'tests/fixtures/*.{vue,[jt]s?(x)}',
    (args, id) =>
      rollupBuild(id, [
        ShortVmodel({ prefix: '::' }),
        RollupVue(),
        UnpluginOxc.rollup(),
      ]),
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
    },
  )
})
