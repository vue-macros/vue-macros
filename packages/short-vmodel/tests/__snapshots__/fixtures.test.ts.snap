// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/basic.vue 1`] = `
"// basic.js
import { defineComponent, ref, resolveComponent, createElementBlock, openBlock, Fragment, createCommentVNode, createVNode } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "basic",
	setup(__props, { expose: __expose }) {
		__expose();
		const val = ref("");
		const __returned__ = { val };
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_A = resolveComponent("A");
  return openBlock(), createElementBlock(
    Fragment,
    null,
    [
      createCommentVNode(" prettier-ignore "),
      createVNode(_component_A, {
        modelValue: $setup.val,
        "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => $setup.val = $event)
      }, null, 8, ["modelValue"]),
      createVNode(_component_A, {
        foo: $setup.val,
        "onUpdate:foo": _cache[1] || (_cache[1] = ($event) => $setup.val = $event)
      }, null, 8, ["foo"])
    ],
    64
    /* STABLE_FRAGMENT */
  );
}
var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], __FILE__]);

export { basic as default };
"
`;
